### ISSUE TITLE
Support for manual vector dimension value assignment of expression

### ISSUE THREAD

**Issue by AmitMY** [clab/dynet#1346]
I'm trying to write a seq2seq with out-of-vocab copy, for which I create a zeros input vector in the size of the out-of-vocab tokens:
```python
copy_vec = dy.inputVector(np.zeros(len(copy_vocab)))
```
But after I value each token seperately (conditional copy mechanism) and have a score for the token, I want to add to that word's scrore so I do:
```python
copy_vec[copy_vocab.w2i[token]] += token_score
```

Which is deriveable as it's a sum, and should just change the expression in that dimension to be a sum of expressions, as far as I understand.

Error:
> TypeError: '_dynet._vecInputExpression' object does not support item assignment

**Comment by pmichel31415** [comment]
Yes this is something that isn't super easy to do right now. One solution is to compute all your token scores separately and concatenate them in `copy_vec` later. Not sure if that's applicable in your case.

**Comment by neubig** [comment]
This is something that we should add though.

**Comment by AmitMY** [comment]
@pmichel31415 that is my current fix, but adding that made my model train 10 times slower on cpu, 60 times slower on gpu.. not 100% sure if it’s this thing or the way I’m calculating the scores, but yeah that sucks
